CHANGELOG.md

```diff
@@ -11,6 +11,12 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ### Removed
 
+## [2.0.0] - 2018-09-28
+
+### Changed
+- Added the `AddContextNewService` and `AddContextService` structs, and deprecated the old `AddContext` struct. One or other of the new structs should be a drop-in replacement for the `AddContext`, depending on whether it was being used as a `NewService` or `Service`.
+- modified the `new_context_type` macro to only implement `Push`, `Pop` and `Has` when the type `T` for types explicitly passed to the macro. This is a breaking change, which should only require minor changes such as adding type annotations if the macro was used as recommended in the docs.
```

Comment:
> modified the `new_context_type` macro to only implement `Push`, `Pop` and `Has` when the type `T` for types explicitly passed to the macro.

Can 'when the type `T`' be removed, perhaps? I don't quite understand what this sentence is trying to say otherwise.

Reply:
Sure, thanks for spotting that!